### DISCUSSION TITLE
v3.0.0-rc.7 and v3.0.0-rc.8

### DISCUSSION THREAD

**Discussion by pi0** [nuxt/nuxt#18389] [Announcements]
# v3.0.0-rc.7 and v3.0.0-rc.8

Release candidates 7 and 8 of Nuxt 3 are out! There has been an extensive amount of work to improve the stability of Nuxt 3 and now we are pleased to announce Vite 3 support :rocket: 

Stay tuned for more exciting stuff and wish you a lovely weekend on behalf of Nuxt team :green_heart: 

### Release notes

- [v3.0.0-rc.7](https://github.com/nuxt/framework/releases/tag/v3.0.0-rc.7)
- [v3.0.0-rc.8](https://github.com/nuxt/framework/releases/tag/v3.0.0-rc.8)

### 🚀 How to upgrade

- **Automated:** `npx nuxi-edge@latest upgrade --force`
- **Manual:** Bump `nuxt` dependency to `3.0.0-rc.8` and then use `npx nuxi-edge@latest cleanup` to cleanup any local caches

**Comment by maxdzin** [comment]
@pi0 Big love 💚 to you and all the Nuxt team for such great work and for keeping updating!

**Comment by durbanitas** [comment]
Is there any indication which of the mentioned [API changes](https://github.com/nuxt/framework/releases/tag/v3.0.0-rc.8) took place? 

Something with my endpoint is not working anymore (used rc.4 before) and I want to dig deeper now. Thanks a lot!

**Reply by pi0** [reply]
Can you please explain more? Changes are indicated with ⚠️ in changelog.

**Reply by durbanitas** [reply]
let me try to explain the current behavior of my feature, but I guess I should update each nuxt version from rc.4 to rc.8 one at a time to figure out what causes the problem.

---

### feature

- get image urls
- compute base64 images on the server side
- send base64 images back to the client

### client

```js
// modify stored dataset
const buildDataset = async (query, ds) => {
  // ...
  try {
    const chunkSize = 25
    const imgUrls = ds.map(img => img.url)
    for (let i = 0; i < dataset.value.length; i += chunkSize) {
      let page = i + chunkSize
      const subset = imgUrls.slice(i, page)
      const images = await getBase64Images(query, subset)
      images.forEach(el => {
        const img = dataset.value.find((img) => img.id === el.id)
        img.base64 = el.base64
      })
    }
  } catch (error) {
    // ...
  }
}

// get base 64 images from the server
const getBase64Images = async (query, urls) => {
  const { data: imgs } = await useFetch(`/api/images?query=${query}`, {
    method: 'post',
    body: { urls }
  })
  return imgs.value
}
```

### server

```js
function base64ArrayBuffer (buffer) {...}

export default defineEventHandler(async (event) => {
  const body = await useBody(event)
  return await Promise.all(
    body.urls.map(async (url) => {
       const obj = {}
        const response = await fetch(url, {
        responseType: 'arrayBuffer',
        headers: {
          mode: 'cors',
          'access-control-allow-origin': '*',
          'content-type': 'image/jpeg'
        }
      })
      const buffer = await response.arrayBuffer()
      const base64 = base64ArrayBuffer(buffer)
      // ... transform data
      return obj 
    })
  )
})
```

----

### problem

in my old version with `rc.4` I can monitor the multiple post requests via vercel's realtime logs, but with the `rc.8` I only see __one__ post request instead of multiple. Because of only one response my app is in a constant loading state.

I only updated to the current nuxt version to `rc.8`, no fancy dependencies are installed, just plain javascript is used.

### question

Is it possible to update the Nuxt version from `rc.4` to `rc.5` and from `rc.5` to `rc.6` and so on?

optional: do you see some mistakes I made?

### update

the `useFetch` post method caches and returns the values from the first call each time. So my constant loading states come from duplicated/missing elements.

I couldn't find anything in the release notes yet, but only found this open issue [#6704](https://github.com/nuxt/framework/issues/6704)

**Reply by durbanitas** [reply]
fixed it for all my `useFetch`-methods with `{ initialCache: false }`

**Comment by max5432112345** [comment]
@pi0 Automated: npx nuxi-edge@latest upgrade --force return

```
Need to install the following packages:
  nuxi-edge@3.0.0-rc.8-27688153.422d5c4
Ok to proceed? (y) y
Nuxt CLI v3.0.0-rc.6-27667209.03cd671                                                                                                                                                11:14:23
ℹ Package Manager: npm 8.15.0                                                                                                                                                        11:14:24
ℹ Current nuxt version: 3.0.0-rc.6-27667209.03cd671                                                                                                                                  11:14:24
ℹ Removing lock-file and node_modules...                                                                                                                                             11:14:24

 ERROR  The "path" argument must be of type string or an instance of Buffer or URL. Received an instance of Promise                                                                  11:14:24

  at Object.rm (node:internal/fs/promises:597:38)
  at node_modules/nuxi/dist/chunks/fs.mjs:11:20
  at Array.map (<anonymous>)
  at rmRecursive (node_modules/nuxi/dist/chunks/fs.mjs:10:27)
  at Object.invoke (node_modules/nuxi/dist/chunks/upgrade.mjs:45:13)
  at async _main (node_modules/nuxi/dist/cli.mjs:46:20)


```

So I removed lock file and installed as always
